### PR TITLE
chore: remove `-mod=readonly` from Makefile

### PR DESCRIPTION
**Pull request by Vikramop**
Closes https://github.com/celestiaorg/celestia-app/issues/2341

### CODE REVIEW

**Review by rootulp — CHANGES_REQUESTED** [review]
Can this PR please remove all instances of `-mod=readonly` in the Makefile?

**Comment by rootulp** [comment]
I updated the PR title to conform to conventional commits: https://www.conventionalcommits.org/en/v1.0.0/

**Comment by Vikramop** [comment]
> Can this PR please remove all instances of `-mod=readonly` in the Makefile?

It remove only the one you have indicated.

**Comment by Vikramop** [comment]
So I've to remove all the -mod=readonly from the makefile.

**Comment by Vikramop** [comment]
Removed all the 7 -mod=readonly.

**Review by rootulp — DISMISSED** [review]

**Comment by rootulp** [comment]
Thanks for the contribution @Vikramop 🎊

**Comment by Vikramop** [comment]
> Thanks for the contribution @Vikramop 🎊

Happy to contribute.

File: Makefile [hunk 1]
```diff
@@ -69,7 +69,7 @@ build: mod
 ## install: Build and install the celestia-appd binary into the $GOPATH/bin directory.
 install: go.sum
 	@echo "--> Installing celestia-appd"
-	@go install -mod=readonly $(BUILD_FLAGS) ./cmd/celestia-appd
+	@go install  $(BUILD_FLAGS) ./cmd/celestia-appd
```
Comment (rootulp) [inline Makefile:72]:
```suggestion
	@go install $(BUILD_FLAGS) ./cmd/celestia-appd
```

File: Makefile [hunk 2]
```diff
@@ -138,33 +138,33 @@ fmt:
 ## test: Run tests.
 test:
 	@echo "--> Running tests"
-	@go test -mod=readonly -timeout 30m ./...
+	@go test  -timeout 30m ./...
```
Comment (rootulp) [inline Makefile:141]:
```suggestion
	@go test -timeout 30m ./...
```

File: Makefile [hunk 3]
```diff
@@ -138,33 +138,33 @@ fmt:
...
 .PHONY: test
 
 ## test-short: Run tests in short mode.
 test-short:
 	@echo "--> Running tests in short mode"
-	@go test -mod=readonly ./... -short
+	@go test  ./... -short
```
Comment (rootulp) [inline Makefile:147]:
```suggestion
	@go test ./... -short
```

File: Makefile [hunk 4]
```diff
@@ -138,33 +138,33 @@ fmt:
...
 .PHONY: test-short
 
 ## test-e2e: Run end to end tests via knuu.
 test-e2e:
 	@echo "--> Running e2e tests"
-	@KNUU_NAMESPACE=celestia-app E2E=true go test -mod=readonly ./test/e2e/... -timeout 30m
+	@KNUU_NAMESPACE=celestia-app E2E=true go test  ./test/e2e/... -timeout 30m
```
Comment (rootulp) [inline Makefile:153]:
```suggestion
	@KNUU_NAMESPACE=celestia-app E2E=true go test ./test/e2e/... -timeout 30m
```

File: Makefile [hunk 5]
```diff
@@ -138,33 +138,33 @@ fmt:
...
 	@echo "--> Running tests in race mode"
-	@go test -mod=readonly ./... -v -race -skip "TestPrepareProposalConsistency|TestIntegrationTestSuite|TestQGBRPCQueries|TestSquareSizeIntegrationTest|TestStandardSDKIntegrationTestSuite|TestTxsimCommandFlags|TestTxsimCommandEnvVar|TestMintIntegrationTestSuite|TestQGBCLI|TestUpgrade|TestMaliciousTestNode|TestMaxTotalBlobSizeSuite|TestQGBIntegrationSuite|TestSignerTestSuite"
+	@go test ./... -v -race -skip "TestPrepareProposalConsistency|TestIntegrationTestSuite|TestQGBRPCQueries|TestSquareSizeIntegrationTest|TestStandardSDKIntegrationTestSuite|TestTxsimCommandFlags|TestTxsimCommandEnvVar|TestMintIntegrationTestSuite|TestQGBCLI|TestUpgrade|TestMaliciousTestNode|TestMaxTotalBlobSizeSuite|TestQGBIntegrationSuite|TestSignerTestSuite"
 .PHONY: test-race
 
 ## test-bench: Run unit tests in bench mode.
 test-bench:
 	@echo "--> Running tests in bench mode"
-	@go test -mod=readonly -bench=. ./...
+	@go test  -bench=. ./...
```
Comment (rootulp) [inline Makefile:167]:
```suggestion
	@go test -bench=. ./...
```

File: Makefile [hunk 6]
```diff
@@ -176,7 +176,7 @@ test-coverage:
 ## txsim-install: Install the tx simulator.
 txsim-install:
 	@echo "--> Installing tx simulator"
-	@go install -mod=readonly $(BUILD_FLAGS) ./test/cmd/txsim
+	@go install  $(BUILD_FLAGS) ./test/cmd/txsim
```
Comment (rootulp) [inline Makefile:179]:
```suggestion
	@go install $(BUILD_FLAGS) ./test/cmd/txsim
```

**Review by rootulp — APPROVED** [review]

**Comment by rootulp** [comment]
Resolved merge conflicts and removed a residual space from each ocurrence.

**Review by rootulp — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]

**Pull request merged**